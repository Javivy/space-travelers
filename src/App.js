import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Header from './Components/Layout/Header/Header';
import Profile from './Components/Pages/Profile/Profile';
import Rockets from './Components/Pages/Rockets/Rockets';
// import Missions from './Components/Pages/Missions/Missions';

function App() {
  return (
    <BrowserRouter>
      <Header />
      <div className="body">
        <Routes>
          <Route path="/" element={<Rockets />} />
          <Route path="/profile" element={<Profile />} />
          {/* <Route path='/missions' element={<Missions />} /> */}
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
