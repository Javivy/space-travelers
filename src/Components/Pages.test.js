import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import App from '../App';

describe('Test the Reservations', () => {
  test('Test the App component', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const linkElement = screen.getByText(/Space Travelers/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('renders the reserved missions page', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Profile'));
    expect(screen.getByText('My Rockets')).toBeInTheDocument();
  });

  test('renders the reserved missions page', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Profile'));
    expect(screen.getByText('My Missions')).toBeInTheDocument();
  });
});

describe('Test the Reservations', () => {
  test('Test the App component', () => {
    const rendered = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
