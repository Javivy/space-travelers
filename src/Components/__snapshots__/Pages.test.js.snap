// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Reservations Test the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="left_section"
        >
          <img
            alt="Logo"
            class="logo"
            src="planet.png"
          />
          <h1
            class="logo_title"
          >
            Space Travelers' Hub
          </h1>
        </div>
        <ul
          class="nav_links"
        >
          <li
            class="nav_link"
          >
            <a
              class=""
              href="/"
            >
              Rockets
            </a>
          </li>
          <div
            class="combine"
          >
            <li
              class="nav_link"
            >
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <div
              class="vLine"
            />
            <li
              class="nav_link"
            >
              <a
                aria-current="page"
                class="active"
                href="/profile"
                style="text-decoration: underline;"
              >
                My Profile
              </a>
            </li>
          </div>
        </ul>
      </nav>
      <div
        class="body"
      >
        <div
          class="profile_bg"
        >
          <div
            class="rockets_bg"
          >
            <div
              class="rockets"
            >
              <h2>
                My Rockets
              </h2>
              <ul>
                <p>
                  You haven't reserved any rockets yet
                </p>
              </ul>
            </div>
          </div>
          <div
            class="missions_bg"
          >
            <div
              class="missions"
            >
              <h2>
                My Missions
              </h2>
              <ul>
                <p>
                  You haven't joined to any mission yet
                </p>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="left_section"
      >
        <img
          alt="Logo"
          class="logo"
          src="planet.png"
        />
        <h1
          class="logo_title"
        >
          Space Travelers' Hub
        </h1>
      </div>
      <ul
        class="nav_links"
      >
        <li
          class="nav_link"
        >
          <a
            class=""
            href="/"
          >
            Rockets
          </a>
        </li>
        <div
          class="combine"
        >
          <li
            class="nav_link"
          >
            <a
              class=""
              href="/missions"
            >
              Missions
            </a>
          </li>
          <div
            class="vLine"
          />
          <li
            class="nav_link"
          >
            <a
              aria-current="page"
              class="active"
              href="/profile"
              style="text-decoration: underline;"
            >
              My Profile
            </a>
          </li>
        </div>
      </ul>
    </nav>
    <div
      class="body"
    >
      <div
        class="profile_bg"
      >
        <div
          class="rockets_bg"
        >
          <div
            class="rockets"
          >
            <h2>
              My Rockets
            </h2>
            <ul>
              <p>
                You haven't reserved any rockets yet
              </p>
            </ul>
          </div>
        </div>
        <div
          class="missions_bg"
        >
          <div
            class="missions"
          >
            <h2>
              My Missions
            </h2>
            <ul>
              <p>
                You haven't joined to any mission yet
              </p>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
